src/System.IO.FileSystem/src/System/IO/CharSpanExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static partial class CharSpanExtensions
+    {
+        public static bool EndsWithOrdinal(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, bool ignoreCase = false)
+        {
+            if (value.Length == 0)
+                return true;
+            else if (value.Length > span.Length)
+                return false;
+
+            if (span.Length != value.Length)
+                span = span.Slice(span.Length - value.Length);
+
+            if (ignoreCase == false)
```

Comment:
this optimization should be inside EqualsOrdinal.

Reply:
It is

Reply:
I mean, so it shouldn't be needed here

Reply:
To Dan's question from a week ago, and related to one of my previous comments, it's odd that EqualsOrdinal checks for it and that we check for it here.  Why are both needed?